src/Button/Button.component.tsx

```diff
@@ -3,113 +3,602 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { darken } from 'polished';
+import { transparentize } from 'polished';
 // THEME
 import { colors, fonts, sizes } from '../theme';
+import { Colors } from '../Typography/Typography.component';
+
+type Variant = 'primary' | 'outline' | 'minimal';
+
+type ButtonSize = 'xlarge' | 'large' | 'small' | 'xsmall';
+
+interface Theme {
+    base: string;
+    light: string;
+    dark: string;
+}
 
 interface ButtonProps {
     children?: any;
     disabled?: boolean;
+    revealed?: boolean;
     showSpinner?: boolean;
     className?: string;
 
+    flip?: boolean;
+    colorTheme?: Theme | Colors;
+    minWidth?: string;
+    block?: boolean;
+    icon?: any;
+    circular?: boolean;
+    reverse?: boolean;
+
+    forceHover?: boolean;
+    forceFocus?: boolean;
+    forceActive?: boolean;
+
     onClick?: () => void;
-    variant?: 'primary' | 'secondary' | 'text';
-    size?: 'small' | 'medium' | 'large';
+    variant?: Variant;
+    size?: ButtonSize;
 }
 
-interface ButtonStylesGroup {
-    [key: string]: FlattenSimpleInterpolation;
+interface StyledButtonProps extends ButtonProps {
+    iconOnly?: boolean;
+    height: number;
+    borderRadius: string;
```

Comment:
Do we want to limit this in any way?

Reply:
It's not a prop on the button itself, so its only used internally

Reply:
As this evolves it could be helpful as a mechanism for communicating the intent of this param to the next dev, but certainly not a requirement.